Modules/Filtering/ImageIntensity/test/itkVectorExpandImageFilterTest.cxx

```diff
@@ -272,57 +272,25 @@ itkVectorExpandImageFilterTest(int, char *[])
     ++streamIter;
   }
 
-  if (!testPassed)
-  {
-    std::cout << "Test failed." << std::endl;
-    return EXIT_FAILURE;
-  }
-
   // Test error handling
+  std::cout << "Setting Input to nullptr" << std::endl;
 
-  try
-  {
-    testPassed = false;
-    std::cout << "Setting Input to nullptr" << std::endl;
-    expander->SetInput(nullptr);
-    expander->Update();
-  }
-  catch (const itk::ExceptionObject & err)
-  {
-    std::cout << err << std::endl;
-    expander->ResetPipeline();
-    expander->SetInput(input);
-    testPassed = true;
-  }
+  expander->SetInput(nullptr);
+  ITK_TRY_EXPECT_EXCEPTION(expander->Update());
 
-  if (!testPassed)
-  {
-    std::cout << "Test failed." << std::endl;
-    return EXIT_FAILURE;
-  }
 
+  expander->ResetPipeline();
+  expander->SetInput(input);
 
-  try
-  {
-    testPassed = false;
-    std::cout << "Setting Interpolator to nullptr" << std::endl;
-    expander->SetInterpolator(nullptr);
-    expander->Update();
-  }
-  catch (const itk::ExceptionObject & err)
-  {
-    std::cout << err << std::endl;
-    expander->ResetPipeline();
-    expander->SetInterpolator(interpolator);
-    testPassed = true;
-  }
+  std::cout << "Setting Interpolator to nullptr" << std::endl;
+  expander->SetInterpolator(nullptr);
+  ITK_TRY_EXPECT_EXCEPTION(expander->Update());
+
+
+  expander->ResetPipeline();
+  expander->SetInterpolator(interpolator);
 
-  if (!testPassed)
-  {
-    std::cout << "Test failed." << std::endl;
-    return EXIT_FAILURE;
-  }
 
   std::cout << "Test passed." << std::endl;
-  return EXIT_SUCCESS;
+  return testPassed;
```

Comment:
`testPassed` is type _bool_, _true_ will turn into 1 (EXIT_SUCCESS is 0).
I think here should be `return EXIT_SUCCESS;` and `if (!testPassed)` block at line 275 should **not** be removed.

Reply:
Thanks for the review @issakomi. Changed the variable type in ed49ad5. Not sure if I see why the block at line 275 should be removed. The subsequent statements only check exceptions (checking exceptions first, and thus, prior to that block in this case, is maybe cleaner in a test).

Reply:
This block should have returned EXIT_FAILURE if _testPassed_ is false. This is no longer applicable. But it would be possible to exit earlier, currently parts of the test runs after _testPassed_ is already false. But it is probably not critical.

Reply:
`testPassed` is never set to `EXIT_SUCCESS` (or `true` in the previous version) by any check once it has been set to `EXIT_FAILURE` by some check (or `false` in the previous version), so there is no risk of an intermediate check turning the value so that the test may report as if it is passing while some previous intermediate check set the flag to failing.

I do not think having removed this block, for example, https://github.com/InsightSoftwareConsortium/ITK/pull/4179/files#diff-7f8f7813216d20a0bc1de2689e2bf80c4994519b44f533e4a6228937ecb8a49bL266 is harmful in that way either; the subsequent checks are not influenced by the results of the checks up to that block.

But let me know if I'm still wrong.

In line with this change, running the test as long as possible even if intermediate checks fail may allow to identify multiple errors at a time. The current ITK SWG advocates for tests running as long as possible. The macros are a caveat to this, but it is the price to pay for a cleaner, more compact and readable test code under the current ITK infrastructure.

Reply:
In the first commit you have deleted the  block `if (!testPassed) return EXIT_FAILURE` and returned boolen testPassed. It is fixed . But parts of test still run after _testPassed_ is already set to 'failed'. It is not very critical and not important for me, honestly, just FYI.

Reply:
I think I only changed the file in a single commit, right? Maybe you refer to the previous commit I amended following your https://github.com/InsightSoftwareConsortium/ITK/pull/4179#discussion_r1311655077. As for the concern in your https://github.com/InsightSoftwareConsortium/ITK/pull/4179#discussion_r1312955726 that is restated in https://github.com/InsightSoftwareConsortium/ITK/pull/4179#discussion_r1313025211, https://github.com/InsightSoftwareConsortium/ITK/pull/4179#discussion_r1312966812 acknowledges that the checks still run after `testPassed` is set to `EXIT_FAILURE` but provides the rationale for that, and highlights that an `EXIT_SUCCESS` is not changed to an `EXIT_FAILURE` value, which would be worrying.

Reply:
I didn't realize that the intention was to run the test "for as long as possible". Fine.